### PR TITLE
Autofind rpc tests --srcdir

### PR DESCRIPTION
**Pull request by jonasschnelli**

### CODE REVIEW

File: qa/rpc-tests/test_framework/test_framework.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def main(self):
                           help="Leave bitcoinds and test.* datadir on exit or error")
         parser.add_option("--noshutdown", dest="noshutdown", default=False, action="store_true",
                           help="Don't stop bitcoinds after the test execution")
-        parser.add_option("--srcdir", dest="srcdir", default="../../src",
+        parser.add_option("--srcdir", dest="srcdir", default=os.path.dirname(os.path.realpath(__file__))+"/../../../src",
```
Comment (MarcoFalke) [inline qa/rpc-tests/test_framework/test_framework.py]: What about adding normpath here to resolve the `../../../`?

```diff
diff --git a/qa/rpc-tests/test_framework/test_framework.py b/qa/rpc-tests/test_framework/test_framework.py
index 039f4e4..0b6d1b6 100755
--- a/qa/rpc-tests/test_framework/test_framework.py
+++ b/qa/rpc-tests/test_framework/test_framework.py
@@ -105 +105 @@ class BitcoinTestFramework(object):
-        parser.add_option("--srcdir", dest="srcdir", default=os.path.dirname(os.path.realpath(__file__))+"/../../../src",
+        parser.add_option("--srcdir", dest="srcdir", default=os.path.normpath(os.path.dirname(os.path.realpath(__file__))+"/../../../src"),

**Comment by MarcoFalke** [comment]
Concept ACK.

**Comment by laanwj** [comment]
Concept ACK.
But should it be `build directory containing bitcoind/bitcoin-cli`? After out-of-tree builds, there is no guarantee that the executables will be where the source is.

**Comment by jonasschnelli** [comment]
I only intended to change the default path for the default build setup. Its just annoying entering `--srcdir=src` all the time. IMO its more convenient to execute the test script from the root build dir instead from the `qa/rpc-tests` dir (which the current default path is referring to).

**Comment by laanwj** [comment]
I agree, just thought maybe we can change the help message at the same time. But not necessary here.

**Comment by MarcoFalke** [comment]
This is indeed a simple and uncontroversial improvement. Using BUILDDIR requires some refactoring, which can be done in a later pull.

@jonasschnelli Mind to fix my above nit?

before nit:

```
  --srcdir=SRCDIR       Source directory containing bitcoind/bitcoin-cli
                        (default: /home/marco/workspace/bitcoin/qa/rpc-
                        tests/test_framework/../../../src)
```

after nit:

```
  --srcdir=SRCDIR       Source directory containing bitcoind/bitcoin-cli
                        (default: /home/marco/workspace/bitcoin/src)

**Comment by jonasschnelli** [comment]
Fixed @MarcoFalke nit.

**Pull request merged**